### DISCUSSION TITLE
Policy Definitions are created as parameter files

### DISCUSSION THREAD

**Discussion by PleaseStopAsking** [Azure/AzOps#677] [Q&A] [answered]
I have read through the documentation and settings details but it is still unclear to me why my `Microsoft.Authorization/policyDefinitions` and `Microsoft.Authorization/policySetDefinitions` that are created from a `Pull` action are created as parameter files and not an actual template as all other resources are within my repo.

Can anyone shed some light on this or point me to documentation that I may have missed?

**Comment by daltondhcp** [comment, marked as answer]
Great question! 
`Microsoft.Authorization/policyDefinitions` and `Microsoft.Authorization/policySetDefinitions` are generated as parameter files and then deployed along with the this ['master template' ](https://github.com/Azure/AzOps/blob/main/src/data/template/template.json). The main reason for this is to avoid having to deal with [json escaping](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-expressions#escape-characters) for more complex policyDefinitions with the deployIfNotExists effect that have templates embedded. 

In the 'legacy' version of AzOps we had a page around this, but realize that is no longer the case. I will make sure we add an item in the FAQ about this.

**Reply by PleaseStopAsking** [reply]
@daltondhcp Thanks for the clarification. Am I correct in assuming that if we wish to add policyDefinitions via PUSH, we should always write them as a parameter file or will the module handle the transformation the first time?

**Reply by daltondhcp** [reply]
You don't have to, it works to deploy/push policies with regular templates as well. The complexity is only there for autogenerated template, if you author your own template you can choose. For consistency purposes a lot of customers choose to do all policies as parameter files though.

**Reply by PleaseStopAsking** [reply]
Thanks for the clarification!
